### PR TITLE
Profile picture v3: Integrated request strategy

### PR DESCRIPTION
**Pull request by vytis**
* Added method that is available to UI from ZMUserSession for setting new profile image
* Added request strategy and upload status class to the operation loop
* Few tweaks for ObjC interoperability

### CODE REVIEW

File: Source/UserSession/UserProfileImageOwner.swift [hunk 1]
```diff
@@ -52,7 +50,7 @@ public final class UserProfileImageOwner: NSObject, ZMImageOwner {
     }
     
     public func originalImageSize() -> CGSize {
-        return imageSize
+        return .zero
```
Comment (mikeger) [inline Source/UserSession/UserProfileImageOwner.swift]: What is the logic behind the change?

File: Source/UserSession/ZMUserSession.m [hunk 2]
```diff
@@ -346,8 +347,8 @@ - (instancetype)initWithTransportSession:(ZMTransportSession *)session
             self.pingBackStatus = [[BackgroundAPNSPingBackStatus alloc] initWithSyncManagedObjectContext:syncManagedObjectContext
                                                                                   authenticationProvider:self.authenticationStatus];
 
-           self.callStateObserver = [[ZMCallStateObserver alloc] initWithLocalNotificationDispatcher:self.localNotificationDispatcher
-                                                                                managedObjectContext:syncManagedObjectContext];
+            self.callStateObserver = [[ZMCallStateObserver alloc] initWithLocalNotificationDispatcher:self.localNotificationDispatcher
```
Comment (mikeger) [inline Source/UserSession/ZMUserSession.m]: 👍

File: Source/UserSession/UserProfileImageOwner.swift (hunk 1 above)
Comment (vytis) [inline Source/UserSession/UserProfileImageOwner.swift]: I don't get the size from UI so I would need to extract it from `NSData`.
I looked through the code and it seems relevant only for assets v2. It's used in https://github.com/wireapp/wire-ios-images/blob/develop/Sources/Image%20Processing/ZMAssetMetaDataEncoder.m. Maybe @daehn can comment more about it?

Comment (daehn) [inline Source/UserSession/UserProfileImageOwner.swift]: Yeah it looks like we do not upload the size with v3 anymore, however I am unsure if the preprocessor uses that information for other purposes.

**Pull request merged**